Add support for ppc64le processor architecture

Sapana-Khemkar opened the pull request:
## Description
We have added support for the ppc64le processor architecture to the dotnet runtime. For that we have added a new value for the System.Runtime.InteropServices.Architecture enum: dotnet/runtime#67428
Code in Microsoft.VisualStudio.TestPlatform.PlatformAbstractions does not recognize this new value. This patch adds a new value to Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.PlatformArchitecture.

## Related issue
NA

Sapana-Khemkar committed:
add support for ppc64le processor architecture

omajid commented on src/Microsoft.TestPlatform.PlatformAbstractions/Interfaces/System/PlatformArchitecture.cs:
```diff
@@ -13,4 +13,5 @@ public enum PlatformArchitecture
     ARM,
     ARM64,
     S390x,
+    PPC64le,
```
Maybe name this `Ppc64le` to keep it consistent with dotnet/runtime?

Sapana-Khemkar replied:
Agree. Will change

omajid commented on src/Microsoft.TestPlatform.PlatformAbstractions/netcore/System/PlatformEnvironment.cs:
```diff
@@ -28,6 +28,7 @@ public PlatformArchitecture Architecture
                 // preview 6 or later, so use the numerical value for now.
                 // case System.Runtime.InteropServices.Architecture.S390x:
                 (Architecture)5 => PlatformArchitecture.S390x,
+                (Architecture)6 => PlatformArchitecture.PPC64le,
```
Is this value correct? Shouldn't this be `8`?  I am looking at https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Architecture.cs, and there's `LoongArch64` and `Armv6` between `S390x` and `PPC64le`.

Sapana-Khemkar replied:
yes you are right. Thanks for catching that. I matched it with enum in VSTest code so counted it 6. Will correct.

Sapana-Khemkar committed:
change enum name from PPC64le to Ppc64le to match with System.Runtime.InteropServices.Architecture enum name

Sapana-Khemkar committed:
add defination of enum Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.PlatformArchitecture.Ppc64le
and Microsoft.VisualStudio.TestPlatform.ObjectModel.Architecture.Ppc64le in Microsoft.TestPlatform.ObjectModel
and /Microsoft.TestPlatform.PlatformAbstractions PublicAPI.Shipped.txt file.

Sapana-Khemkar committed:
corrected typo in ppc64le arch name

leecow commented:
@nohwnd - could you have a look so that we can queue this up for .NET 7 GA review?

cc @crummel

Sapana-Khemkar commented:
@nohwnd  Thanks for merging. Where can I find daily build artifacts for VSTest repo? Azure Link provided in readme is not reachable. Can you please help?

nohwnd commented:
@Sapana-Khemkar there is some issue with our token for publishing into our public feed, we are trying to fix it so you can see the preview package that would go into net8 and possibly try it.

Sapana-Khemkar commented:
> @Sapana-Khemkar there is some issue with our token for publishing into our public feed, we are trying to fix it so you can see the preview package that would go into net8 and possibly try it.

Noted. Thanks.

Sapana-Khemkar commented:
@nohwnd It seems Azure link is yet not up. Is there anyway to download daily build binary for testing?

janani66 commented:
@nohwnd  - Where can we download the preview package that would go into net8? Just trying to access a binary build with the fix for #4028 for testing purposes.

nohwnd commented:
@Sapana-Khemkar  @janani66 Just using the latest dotnet installer for net8-alpha should be the easiest way to acquire it.  The latest successful merge from vstest to dotnet/sdk happened 3 days ago, and from sdk to installer 2 days ago. So it should be safely there. But please double check the version printed in the console when you use `dotnet test`, the number is a date, you want October 1st or newer. 


As for the nuget package we insert into dotnet/sdk that one is here: https://dev.azure.com/dnceng/public/_artifacts/feed/test-tools/NuGet/Microsoft.TestPlatform.CLI/overview/17.5.0-preview-20221007-01 

If you want redistributable .NET Framework package you should choose Microsoft.TestPlatform (without the .CLI) instead. That is what CI servers use to run tests, when not using `dotnet test`.

This is the feed we push it into: https://dev.azure.com/dnceng/public/_artifacts/feed/test-tools

nohwnd commented:
@janani66 Or maybe you want Microsoft.NET.Test.SDK which is what you install in your .NET project, that package is here https://dev.azure.com/dnceng/public/_artifacts/feed/test-tools/NuGet/Microsoft.NET.Test.Sdk/overview/17.5.0-preview-20221007-01 on the same feed.

janani66 commented:
The test with the 17.5.0-preview-20221007-01 failed to recognize ppc64le as a supportedArchitecture. Has something changed In .NET7 w.r.t. VSTest integration with .NET . 

Error snippet:
[ERROR] FATAL UNHANDLED EXCEPTION: System.NotSupportedException: Specified method is not supported.
   at Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.ProcessHelper.GetCurrentProcessArchitecture()
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Executor.PrintSplashScreen(Boolean isDiag)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Executor.Execute(String[] args)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Program.Run(String[] args, UiLanguageOverride uiLanguageOverride)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Program.Main(String[] args)

-------------
Am running the roslyn tests with .net7 rc1 where I put the downloaded 17.5.0-preview-20221007-01 nuget file
in a local directory and pointed to it via the following in the NuGet.config file (Note I do see the 17.5.0-preview-20221007-01 dll being referenced by dotnet exec) 

<add key="local-ppc64le" value="/home/ubuntu/sapana/preview7/output" />

Also added this to the eng/Versions.xml  to override the default MicrosoftNETTestSdkVersion
<MicrosoftNETTestSdkVersion>17.5.0-preview-20221007-01</MicrosoftNETTestSdkVersion>


-----------
ubuntu@dotnet1:~/janani$ dotnet --info
.NET SDK:
 Version:   7.0.100-preview.7.22377.5
 Commit:    ba310d9309

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         linux-ppc64le
 Base Path:   /home/ubuntu/sapana/preview7/output/.dotnet/sdk/7.0.100-preview.7.22377.5/

Host:
  Version:      7.0.0-preview.7.22375.6
  Architecture: ppc64le
  Commit:       N/A

.NET SDKs installed:
  7.0.100-preview.7.22377.5 [/home/ubuntu/sapana/preview7/output/.dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.0-preview.7.22376.6 [/home/ubuntu/sapana/preview7/output/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.0-preview.7.22375.6 [/home/ubuntu/sapana/preview7/output/.dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/home/ubuntu/sapana/preview7/output/.dotnet]

global.json file:
  Not found

MarcoRossignoli commented:
@janani66 the new enum won't be shipped into 7.0 you should try with the preview SDK 8.0 that you download from here https://github.com/dotnet/installer#table

janani66 commented:
Looks like we also need the fix : https://github.com/microsoft/vstest/pull/4066

janani66 commented:
I was able to test with the 17.5.0-preview-20221024-01 build published on https://pkgs.dev.azure.com/dnceng/public/_packaging/test-tools/nuget/v3/index.json. and the dotnet xunit test passed.

A couple of comments/questions:

- If I add a nuget.config file like shown at: https://dev.azure.com/dnceng/public/_artifacts/feed/test-tools/connect/nuget.exe

it is not able to find "Newtonsoft.Json": "13.0.1". 

Once I remove the "clear" from the nuget.config, it found NewtonSoft.Json from nuget.org. Is there a reason why the example nuget.config on the dancing feed shows the clear in the example nuget.config?

i.e. 
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <clear />
    <add key="test-tools" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/test-tools/nuget/v3/index.json" />
  </packageSources>
</configuration>
```

- Just wanted to point out that on this page https://dev.azure.com/dnceng/public/_artifacts/feed/test-tools/connect/dotnet, the {0}/{1}/{2}  variables are not substituted  for nuget.config {0}/.csproj{1}/.sln{{2}

MarcoRossignoli commented:
> Is there a reason why the example nuget.config on the dnceng feed shows the clear in the example nuget.config?

Because our test-tools feed doesn't have any upstream and so there you can find only packages released by us and not for instance `Newtonsoft.Json": "13.0.1`

> Just wanted to point out that on this page https://dev.azure.com/dnceng/public/_artifacts/feed/test-tools/connect/dotnet, the {0}/{1}/{2} variables are not substituted for nuget.config {0}/.csproj{1}/.sln{{2}

this is not clear to me...can you elaborate? I don't see any `{0}/{1}/{2}` in that page.

janani66 commented:
> > Is there a reason why the example nuget.config on the dnceng feed shows the clear in the example nuget.config?
> 
> Because our test-tools feed doesn't have any upstream and so there you can find only packages released by us and not for instance `Newtonsoft.Json": "13.0.1`
> 
> > Just wanted to point out that on this page https://dev.azure.com/dnceng/public/_artifacts/feed/test-tools/connect/dotnet, the {0}/{1}/{2} variables are not substituted for nuget.config {0}/.csproj{1}/.sln{{2}
> 
> this is not clear to me...can you elaborate? I don't see any `{0}/{1}/{2}` in that page.

On the https://dev.azure.com/dnceng/public/_artifacts/feed/test-tools/connect/dotnet page, I see under "Project setup"

**Add a {0} file to your project, in the same folder as your {1} or {2} file**

<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <clear />
    <add key="test-tools" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/test-tools/nuget/v3/index.json" />
  </packageSources>
</configuration>
